improver/utilities/common_input_handle.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown copyright. The Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+from typing import Union
+
+from iris.cube import Cube, CubeList
+
+from improver.utilities.flatten import flatten
+
+
+def as_cubelist(*cubes: Union[Cube, CubeList]):
+    """
+    Standardise input handling of cube/cubelist arguments.
+
+    The role of this function is to flatten the provided inputs and thereby
+    return a single CubeList object.
+
+    Args:
+        cubes:
+            Input data provided in the form of one or more cubes or cubelists (or mixture thereof).
+
+    Returns:
+        CubeList:
+            A CubeList containing all the cubes provided as input.
+    """
+    if not cubes or all([not cube for cube in cubes]):
+        raise ValueError("One or more cube should be provided.")
+    return CubeList(flatten(cubes))
+
+
+def as_cube(cube: Union[Cube, CubeList]):
+    """
+    Standardise input handling of cube arguments.
+
+    The role of this function is to return a single cube object.
+
+    Args:
+        cube:
+            Input data provided in the form of a cube ir cubelists.
+
+    Returns:
+        Cube:
+            A single cube.
+    """
+    if not cube:
+        raise ValueError("A cube should be provided.")
+    if isinstance(cube, CubeList):
```

Comment:
if not a cube or cubelist an exception is raised to say that a cube must be provided

Reply:
I was wondering if you wanted to handle `[cube1, cube2]`, as well as `iris.cube.CubeList([cube1, cube2])`.

Reply:
well `as_cube` would currently throw an exception for the case where a cubelist is provided containing more than 1 cube (ultimately one could extend to internally call merge_cube perhaps or provide a UI to do this).

I saw `as_cube` as taking the role of returning a single cube (raising an exception where this isn't possible).
I saw `as_cubes` as taking the role of returning a single cubelist.

Motivated by the fact that I think that any function or class should needn't hardcode an expectation to be provided a cube, since a cubelist containing a single cube should be equally suitable.

Retuning more than 1 cube, well I think that's the job of the cubelist, hence why I thought `as_cube` would be best placed returning only the single cube case.  That is, I think `as_cubes([cube1, cube2])` would be better to use no??